lib/spack/spack/platforms/test.py

```diff
@@ -38,8 +38,12 @@ class Test(Platform):
 
     def __init__(self):
         super(Test, self).__init__('test')
-        self.add_target(self.default, Target(self.default))
-        self.add_target(self.front_end, Target(self.front_end))
+
+        default_target = Target(self.default, "x86_64")
```

Comment:
Request: make it clearer that `x86_64` is a name for a module file; IMO replacing with `/path/to/backend_module` would be clearer. I think it's also worth adding a comment that the test module files don't exist (and also potentially mention that this is the reason for `mock_modulecmd`).